smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/MovePlanMaker.java

```diff
@@ -311,26 +313,16 @@ class StorageTypeDiff {
      * @returns the remaining number of replications to move.
      */
     int removeOverlap(boolean ignoreNonMovable) {
-      for(Iterator<String> i = existing.iterator(); i.hasNext(); ) {
-        final String t = i.next();
-        if (expected.remove(t)) {
-          i.remove();
-        }
-      }
+      existing.removeIf(expected::remove);
       if (ignoreNonMovable) {
```

Comment:
Consider refactoring this block of code because it currently has a very non-obvious behavior